### PR TITLE
Fix func-test-pr handling to support stable branches

### PR DESCRIPTION
**Pull request by hemanthnakkina**
Substitution of git url links in test-requirements with
func-test-pr does not consider branch name in git url.
This patch adds stable branch regex to consider for
substitution.

### CODE REVIEW

**Comment by gnuoy** [comment]
Thank you for this PR, it is very much appreciated. I would really like to see some unit tests before this lands. I know that is asking a lot given they don't exist at the moment so I would suggest we put this PR on hold until some are added (I'll try and put a pr up for that later today)

**Comment by gnuoy** [comment]
Could you rebase your branch and extend the unit tests to cover your change please.

**Comment by hemanthnakkina** [comment]
@gnuoy rebased the branch and modified the unit testcases

**Review by gnuoy — APPROVED** [review]
Great work, thank you

**Pull request merged**
